### PR TITLE
Natchez datadog

### PR DESCRIPTION
**Pull request by tomverran**
This adds an integration between [Natchez](https://github.com/tpolecat/natchez) and Datadog. It includes an OVO Specific quirk in that we pass around trace tokens as well as Datadog trace IDs because Datadog IDs must be numeric and none of our existing trace tokens are.

The gist is that the effect type we deal with now is a `Kleisli[F, Span[F], A]` rather than the old `StateT` and the `Span[F]` contains `Ref`s to enable you to set/get tags and also provides functions to create subspans. All the span creation methods return a `Resource[F, Span[F]]` that on close submits the completed span to a queue which periodically gets sent to the Datadog agent

### CODE REVIEW

File: logging/src/main/scala/com/ovoenergy/effect/Tracing.scala [hunk 1]
```diff
@@ -66,29 +66,29 @@ object Tracing {
...
 
-  def mdc[F[_] : Tracing: Functor]: F[Tags] =
+  def mdc[F[_]: Tracing: Functor]: F[Tags] =
     Tracing[F].get.map(c => c.mdc.updated("traceToken", c.token.value))
 
   def trace[F[_]: FlatMap](fn: Tags => F[Unit])(implicit t: Tracing[Traced[F, ?]]): Traced[F, Unit] =
     mdc[Traced[F, ?]].flatMapF(fn)
 
-  def putMdc[F[_] : Tracing : Monad](tags: (String, String)*): F[Unit] =
+  def putMdc[F[_]: Tracing: Monad](tags: (String, String)*): F[Unit] =
```
Comment (gregorath) [inline logging/src/main/scala/com/ovoenergy/effect/Tracing.scala]: what have you done, did you update your local settings or it was always like this but the rules werent applied?

Comment (tomverran) [reply logging/src/main/scala/com/ovoenergy/effect/Tracing.scala]: I think we just haven't run scalafmt on the project for ages 🤷‍♂

**Comment by mwz** [comment]
Have you seen https://github.com/Colisweb/scala-opentracing by any chance? I'm curious how it compares against Natchez...

**Comment by tomverran** [comment]
I did only very briefly, this has been a self indulgent learning exercise 😄 - Natchez is at its core just a couple of type classes we implement here

**Pull request merged**